source/elements/oneMKL/source/domains/dft/config_params/data_layouts.rst

```diff
@@ -109,20 +113,35 @@ as expressed in :eq:`eq_idx_data_layout` corresponding to two different
 :math:`(d+1)`-tuples :math:`(m, k_{1}, k_{2}, \dots, k_d)` within valid range,
 *i.e.*, such that
 
+.. _onemkl_dft_data_layout_nonredundant_valid_range:
+
 - :math:`0 \leq m < M` wherein :math:`M` is the :ref:`number of transforms<onemkl_dft_config_number_of_transforms>`;
 
-- :math:`0 \leq k_j < n_j`, :math:`1 \leq j < d`;
+- :math:`0 \leq k_j < n_j`, :math:`1 \leq j < d` if :math:`d > 1` ;
 
 - :math:`0 \leq k_d < n_d`, for descriptors of ``COMPLEX`` template :ref:`onemkl_dft_enum_domain` or for forward data of descriptors of ``REAL`` template :ref:`onemkl_dft_enum_domain`;
 
 - :math:`0 \leq k_d \leq \lfloor\frac{n_d}{2}\rfloor`, for backward data of descriptors of ``REAL`` template :ref:`onemkl_dft_enum_domain`.
 
-Additionally, for in-place transforms (``PLACEMENT=INPLACE``), the forward and backward data
-layouts must be configured *consistently* with one another, that is,
-
-- for descriptors of ``COMPLEX`` template :ref:`onemkl_dft_enum_domain`, strides and distances must be equal in forward and backward domain;
-
-- for descriptors of ``REAL`` template :ref:`onemkl_dft_enum_domain` with :ref:`onemkl_dft_real_storage_real_real` and :ref:`onemkl_dft_conjugate_even_storage_complex_complex`, considering any :math:`(d+1)`-tuple :math:`(m, k_{1}, k_{2}, \dots, k_{d-1}, 0)` within valid range (as explicited above), the index value :eq:`eq_idx_data_layout` in forward domain must be twice the corresponding index value for the same tuple in backward domain.
+Additionally, for in-place transforms (``PLACEMENT=INPLACE``), the forward and
+backward data layouts must be configured *consistently* with one another: the
+memory address(es) of leading element(s) along the last dimension must be
```

Comment:
Is the last dimension always the one this should be about? Or should it be the dimension with the smallest stride? And I think there should be a requirement that the same dimension has the smallest stride in both domains.

The special cases for existing formats do clarify that, but if we have a general description it should specify the same.

Reply:
Yes, it is always the last dimension, this is now explicitly specified in section "Finite range of indices" of the landing page (title "Discrete Fourier Transform Functions").

Your comment about "same dimension for smallest stride" is indeed required for in-place transforms (real and complex), *e.g.*, to clarify that something like $\text{fwdstrides} = {0, 2(B/2+1), 2A(B/2+1)}$ and $\text{fwdstrides} = {0, (B/2+1), 1}$ for real in-place is not considered consistent either.

I have just addressed that in [commit bf0048e3](https://github.com/oneapi-src/oneAPI-spec/pull/506/commits/bf0048e32b5b2cde0e1c1c0703e267f8bb2667d2). Please let me know if you agree with that (thumbs up) or comment.

Reply:
Looks better. 

However, I still have doubts in this always being about the last dimension. What you are proposing right now, for example, allows the following configuration:
real transform, inplace, size: [2,3,5], FWD_STRIDES: [0,1,2,12], BWD_STRIDES: [0,1,2,6]

I assumed that the intention of this requirement was to disallow cases like that that can not be processed line by line (where by line I mean the consecutive elements in the dimension with the smallest stride), as doing it that way would partially overwrite the data not yet processed. Feel free to resolve if that is not the case.

Reply:
No, it does not because your example uses $d = 3$ and does not satisfy the condition $\text{fwdstrides}\left(\ell\right) = 2\text{bwdstrides}\left(\ell\right)$ for $\ell \in {1, 2}$ (see section "General consistency requirements" in page "Configuration of Data Layouts"), which is precisely meant for ensuring what you point out.

Real transform, inplace, size: [2,3,5], with FWD_STRIDES: [0,**2,4,12**], BWD_STRIDES: [0,1,2,6] can be supported though and since only **every other** real element is meaningful in the forward-domain data container in that case, the relevant data can't be overwritten.